pytorch_lightning/trainer/connectors/profiler_connector.py

```diff
@@ -11,16 +11,30 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License
-from pytorch_lightning.profiler import PassThroughProfiler, SimpleProfiler
+
+from typing import Union
+
+from pytorch_lightning.profiler import BaseProfiler, PassThroughProfiler, SimpleProfiler, AdvancedProfiler
+from pytorch_lightning.utilities import rank_zero_warn
 
 
 class ProfilerConnector:
 
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def on_trainer_init(self, profiler):
+    def on_trainer_init(self, profiler: Union[BaseProfiler, bool, str]):
+
+        if isinstance(profiler, bool):
+            rank_zero_warn("Passing a bool value as a `profiler` argument to Trainer is deprecated"
+                           " and will be removed in v0.11.0. Use str ('simple' or 'advanced') instead.",
+                           DeprecationWarning)
         # configure profiler
-        if profiler is True:
+        if profiler is True or profiler == "simple":
             profiler = SimpleProfiler()
+        elif profiler == "advanced":
+            profiler = AdvancedProfiler()
+        elif isinstance(profiler, str):
```

Comment:
Shall we also not allow other types?

Reply:
yeah I think we can have a condition above
```python
if not isinstance(profiler, (bool, str, BaseProfiler)):
    raise MisconfigurationException(...)
```

Reply:
or rather replace
```py
elif isinstance(profiler, str):
```
with
```py
if not isinstance(profiler, (bool, 'simple', 'advanced', BaseProfiler)):
    raise MisconfigurationException(...)
```
if we only allow `simple` and `advanced`

Reply:
@ydcjeff Are you sure the `isinstance` construct is valid? Python 3.8 throws me a `TypeError`:
`>>> isinstance("abcd", (bool, "abcd"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type or tuple of types
`

Reply:
@ydcjeff `'simple', 'advanced'` this is not data type...

Reply:
Ahh yeah, sorry for that

Reply:
Can we have a test for passing other types than string?

Reply:
Added a test for incorrect datatypes (int, float, dict, list, tensor, Trainer) for which a `MisconfigurationException` is raised. A previous test already checks that the Profiler is created as expected for correct argument types.